### ISSUE TITLE
Add a nuxt sidebase setup wizard

### ISSUE THREAD

**Issue by zoey-kaiser** [sidebase/sidebase#13]
### Describe the feature

Currently when someone wants to use the scaffold they have to clone the repo. We could create a setup wizard with `npx` that automatically sets up the project locally for the user. This could also solve #2, as we can offer a selection of CSS frameworks that the user can choose from. 

We could also let the user customize how much they want to include in their setup (do they want a backend? Do they want a component testing system? Do they want JEST?). 

This could allow it to be come even more versatile in the future and truly become THE SCAFFOLD for nuxt 3.

**Comment by BracketJohn** [comment]
Yeah, that's a great one!

It was already raised by @valiafetisov during the review here: https://github.com/sidestream-tech/sidebase/pull/1
I'vew also opened an issue to extrract the remaining suggestions from nuxt/nuxt#1 into issues here: nuxt/nuxt#8


Some stuff from duplicate nuxt/nuxt#25:
- what css component libraries do you want to use?
- do you want to have a backend?
- vite or webpack?
- ...

Example of where we once build a cli: https://github.com/sidestream-tech/unified-auctions-ui/pull/402

some libraries to use:
- prompts


Related issues in nuxt-repo:
- https://github.com/nuxt/nuxt.js/issues/13552
- https://github.com/nuxt/nuxt.js/issues/13550
- https://github.com/nuxt/framework/discussions/3986
  - here @atinux referred to framework/discussions/569 with: "This is why we did not migrate create-nuxt-app to Nuxt 3."
- https://github.com/nuxt/framework/discussions/569
  - outcome: make `nuxi` support this, but may be hard to support as `create-nuxt-app` was already hard to support

Related:
- https://github.com/nuxt/create-nuxt-app
- discussion in `create-nuxt-app` repo: https://github.com/nuxt/create-nuxt-app/discussions/927

**Comment by BracketJohn** [comment]
@Atinux / @pi0 / @danielroe as we don't want to re-invent the wheel + this seems like something you as the nuxt-team has probably planned in:
- Is there an open issue / an ongoing plan to build a CLI for project setup, as existed for nuxt 2?
- Would you be happy to have this done by the community or rather not?

We were only able to find the two issues linked above, which are rather about "project structure" than about "do you want to use storybook, histoire or vitebook?".

**Comment by zoey-kaiser** [comment]
> If these are going into branches, may I please suggest into keeping them without lock-files so that any dependency manager can be used? I was looking into the logistics of this given the current structure of the project where the root of the default branch is the replica of the template, while now I would suggest that since the development also takes place on that branch, we setup a pipeline where `dev` is the development branch for making changes to the template and the newest commits reflect on `main` without the lock-file but would be used as the starter template.
> 
> Now that I'm typing this out, I feel this could be a separate issue!

Hi @ineshbose 👋 

I really like the idea! Would you mind opening an issue for us, so we can work on it and keep you in the loop? Thank you so much!

**Comment by ineshbose** [comment]
May I ask where would be an ideal place to list the questions/options for this? I'm hoping that one possible option can be using Prisma instead of TypeORM!

Regarding the implementation as well, if these are going on as branches, would it create a large amount of branches given the combinations? Would it instead install/add them to the `package.json`? But that would then require manual configuration and integration of the package to Nuxt!

**Comment by zoey-kaiser** [comment]
> May I ask where would be an ideal place to list the questions/options for this? I'm hoping that one possible option can be using Prisma instead of TypeORM!

I will be creating a new issue once we get to the point where we want to start listing options!

> Regarding the implementation as well, if these are going on as branches, would it create a large amount of branches given the combinations? Would it instead install/add them to the `package.json`? But that would then require manual configuration and integration of the package to Nuxt!

We are still discussing which solution we want to end up using. If possible we want to avoid having a thousand branches with slight modifications. We are currently looking into what the best possible solution would be to add single packages and their configuration into one big final project!
If you have any experience or ideas, feel free to share them!

**Comment by BracketJohn** [comment]
Collection of options we _may_ support.

#### ORMs

- TypeORM
- Prisma

#### CSS utility libraries

- TailwindCSS
- UnoCSS

#### CSS Component libraries

- Ant Design Vue
- Naive UI
- Oruga
- PrimeVue
- Quasar
- Element UI
- Agnostic UI
- Flowbite Vue
- Vuetify


-> Let's expand this over time

**Comment by mahdiboomeri** [comment]
> Let's expand this over time

Hey, it'd be great if we could add tRPC to this list as I previously suggested on [Twitter](https://twitter.com/mahdi_boomeri/status/1567580846151925762?s=20&t=6p7IWi6D6yUfecHaqUysoQ), I'm also happy to help in the implementation if we decided to move forward with it.

**Comment by zoey-kaiser** [comment]
Discussion continued in #59

**Comment by BracketJohn** [comment]
Closing this for now: We made a PoC of this here: https://github.com/sidebase/create-nuxt3-app

We tested out a bunch of different approaches and it all confirms what the nuxt-team has already learned / pointed out for `create-nuxt-app`: It's quite hard to maintain. For now we're dropping the plans to create `create-nuxt3-app`.

If you'd like to have the repo / package name feel free to reach out via DM on discord https://discord.gg/9MUHR8WT9B, open an issue here or open an issue in https://github.com/sidebase/create-nuxt3-app

Closing this for now.
